src/backgroundtasks.cs

```diff
@@ -0,0 +1,99 @@
+//
+// BackgroundTasks C# bindings
+//
+// Authors:
+//	Manuel de la Pena Saenz <mandel@microsoft.com>
+//
+// Copyright 2019 Microsoft Corporation All rights reserved.
+//
+using System;
+
+using CoreFoundation;
+using Foundation;
+using ObjCRuntime;
+
+namespace BackgroundTasks {
+
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof(BGTaskRequest))]
+	interface BGAppRefreshTaskRequest {
+		[Export ("initWithIdentifier:")]
+		IntPtr Constructor (string identifier);
+	}
+
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof(BGTaskRequest))]
+	interface BGProcessingTaskRequest {
+		[Export ("initWithIdentifier:")]
+		IntPtr Constructor (string identifier);
+
+		[Export ("requiresNetworkConnectivity")]
+		bool RequiresNetworkConnectivity { get; set; }
+
+		[Export ("requiresExternalPower")]
+		bool RequiresExternalPower { get; set; }
+	}
+
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface BGTaskRequest : NSCopying {
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[NullAllowed, Export ("earliestBeginDate", ArgumentSemantic.Copy)]
+		NSDate EarliestBeginDate { get; set; }
+	}
+
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface BGTask
+	{
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[NullAllowed, Export ("expirationHandler", ArgumentSemantic.Strong)]
+		Action ExpirationHandler { get; set; }
+
+		[Export ("setTaskCompletedWithSuccess:")]
+		void SetTaskCompleted (bool success);
+	}
+
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof(BGTask))]
+	interface BGAppRefreshTask {
+	}
+
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof(BGTask))]
+	interface BGProcessingTask {
+	}
+
+	[TV (13,0), NoWatch, NoMac, iOS (13,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface BGTaskScheduler {
+		[Static]
+		[Export ("sharedScheduler", ArgumentSemantic.Strong)]
+		BGTaskScheduler SharedScheduler { get; }
+
+		[Advice ("This API is not available when using iOS App Extensions.")]
+		[Export ("registerForTaskWithIdentifier:usingQueue:launchHandler:")]
+		bool RegisterForTask (string identifier, [NullAllowed] DispatchQueue queue, Action<BGTask> launchHandler);
+
+		[Export ("submitTaskRequest:error:")]
+		bool Submit (BGTaskRequest taskRequest, [NullAllowed] out NSError error);
```

Comment:
Not sure if I'm happy about removing the TaskRequest part of the name. It look redundant, but I'm not sure.

Reply:
I would vote to keep `SubmitTaskRequest` brings more context to the action you want to perform

Reply:
In doubt look an the swift name

Reply:
@spouliot swift uses the same name pattern :)